fund.go

```diff
@@ -0,0 +1,134 @@
+package fineractor
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	rh "github.com/tsocial/credit-line-common/retryable-http"
+)
+
+type FundIncrementRequest struct {
+	Id                string `json:"-"`
+	Locale            string `json:"locale"`
+	DateFormat        string `json:"dateFormat"`
+	TransactionDate   string `json:"transactionDate"`
+	TransactionAmount string `json:"transactionAmount"`
+	PaymentTypeId     string `json:"paymentTypeId"`
+}
+
+type FundIncrementResponse struct {
+	OfficeId   float64
+	ClientId   float64
+	ResourceId float64
+}
+
+type FundDecrementRequest struct {
+	Id                string `json:"-"`
+	Locale            string `json:"locale"`
+	DateFormat        string `json:"dateFormat"`
+	TransactionDate   string `json:"transactionDate"`
+	TransactionAmount string `json:"transactionAmount"`
+	PaymentTypeId     string `json:"paymentTypeId"`
+}
+
+type FundDecrementResponse struct {
+	OfficeId   float64
+	ClientId   float64
+	ResourceId float64
+}
+
+type FundValueRequest struct{}
+
+type FundValueResponse struct{}
+
+type FundAvailablityRequest struct{}
+
+type FundAvailablityResponse struct{}
+
+type FundsRequest struct{}
+
+type FundsResponse struct{}
+
+func (client *Client) FundIncrement(request FundIncrementRequest) (FundIncrementResponse, error) {
+	b, err := json.Marshal(request)
+	if err != nil {
+		log.Println(err)
+		return FundIncrementResponse{}, err
+	}
+	req, err := http.NewRequest("POST", client.HostName+request.Id+"/transactions?command=deposit", bytes.NewBuffer(b))
+	if err != nil {
+		log.Println(err)
+		return FundIncrementResponse{}, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("fineract-platform-tenantid", "default")
+	// TODO: Add a better way to generate this authorisation
+	req.Header.Set("Authorization", "Basic bWlmb3M6cGFzc3dvcmQ=")
```

Comment:
This cannot be hard coded.

Reply:
How do I get this auth token? Hence I have added a #TODO, I need a better way